java-manta-client-unshaded/src/main/java/com/joyent/manta/client/MantaClient.java

```diff
@@ -313,66 +325,133 @@ public void delete(final String rawPath) throws IOException {
     public void deleteRecursive(final String path) throws IOException {
         LOG.debug("DELETE {} [recursive]", path);
 
-        try {
-            this.delete(path);
-            LOG.debug("Finished deleting path {}", path);
-            return;
-        } catch (MantaClientHttpResponseException e) {
-            /* In the case of the directory not being empty, we let
-             * that error case go uncaught and pass through
-             * because it is in fact a directory with files. */
-            if (!e.getServerCode().equals(DIRECTORY_NOT_EMPTY_ERROR)) {
-                throw e;
-            }
-        }
-
-        try (Stream<MantaObject> objects = this.listObjects(path)) {
-            objects.forEach(obj -> {
-                try {
-                    if (obj.isDirectory()) {
-                        this.deleteRecursive(obj.getPath());
-                    } else {
-                        this.delete(obj.getPath());
+        /* We repetitively run the find() -> delete() stream operation and check
+         * the diretory targeted for deletion by attempting to delete it this
+         * deals with unpredictable directory contents changes and occasional
```

Comment:
"unpredictable directory contents" 

Is this referring to concurrent modifications?

Reply:
Yes, it is. I will write that specifically.